File: replication/repl_info.rb
require 'rubygems'
require 'mongo'

include Mongo
master = Connection.new('localhost', 27017)
slave  = Connection.new('localhost', 9999, :slave_ok => true)

master_local = master.db('local')
slave_local  = slave.db('local')

master_oplog = master_local["oplog.$main"].find.sort([["ts", -1]]).to_a.first

slave_oplog = slave_local.collection("sources").find_one

diff = master_oplog['ts'][1] - slave_oplog['syncedTo'][1]

puts "Slave is behind master by #{diff} seconds"

File: replication/repl_info.py

# this computes and prints how far behind a slave is from a master

from pymongo.connection import Connection
from pymongo import ASCENDING, DESCENDING

slave = Connection( "localhost" , 9999 , slave_okay=True )

source = slave["local"]["sources"].find_one()
lastSyncedSeconds = source["syncedTo"][1]
print( source )

master = Connection( source["host"] )

oplog = master["local"]["oplog.$main"]
lastOp = oplog.find().limit(1).sort( "ts" , DESCENDING )[0]
lastOpSeconds = lastOp["ts"][1]
print( lastOp )

diffSeconds = lastOpSeconds - lastSyncedSeconds
print( "slave is behind by: %s seconds" % diffSeconds )




